plan/column_pruning.go

```diff
@@ -30,13 +31,29 @@ func getUsedList(usedCols []*expression.Column, schema expression.Schema) []bool
 	return used
 }
 
+func exprCanPrune(expr expression.Expression) bool {
```

Comment:
I think explicitly name it as `exprHasSetVar` is better.
This function doesn't consider if expr is used by parent, 
this returns true doesn't mean it can be prune.